### PR TITLE
Fix import channel in Postgresql 

### PR DESCRIPTION
**Pull request by jredrejo**
## Summary
When importing channel data, [psycopg2 execute_values](https://www.psycopg.org/docs/extras.html#psycopg2.extras.execute_values) function was used. This [function code](https://github.com/psycopg/psycopg2/blob/master/lib/extras.py#L1296)  converts to _bytes_ all the _strings_ in order to have a better performance.
However,  converting an utf-8 _char_ to _byte_ results in more than one byte, making some strings unfit in the maximum number of chars of a column limit.

This PR:
- Replaces the use of `execute_values` by `executemany`
- As sqlite does not applies the column char limit, ensures limit is applied to data before is inserted in Postgresql

Note: I had to cherry-pick the commit from  https://github.com/learningequality/kolibri/pull/12466 to fix docs builds in GH

## References
Closes: #11780

## Reviewer guidance
Do tests pass?
In order to test the fix, this channel was failing and can be used in a kolibri installation using PG:
`kolibri manage importchannel network --baseurl=https://studio.learningequality.org 07cd1633691b4473b6fda08caf826253`

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/content/utils/channel_import.py [hunk 1]
```diff
@@ -647,13 +654,11 @@ def generate_data_with_default(record):
                         )
                     )
                 else:
-                    execute_values(
-                        cursor,
-                        # We want to overwrite new values that we are inserting here, so we use an ON CONFLICT DO UPDATE here
-                        # for the resulting SET statement, we generate a statement for each column we are trying to update
-                        "INSERT INTO {table} AS SOURCE ({column_names}) VALUES %s ON CONFLICT ({pk_name}) DO UPDATE SET {set_statement};".format(
+                    cursor.executemany(
```
Comment (rtibbles) [inline kolibri/core/content/utils/channel_import.py]: This will cause a considerable slowdown in insert performance https://stackoverflow.com/a/39034789 so I'd like us to find another way to handle this.

**Review by rtibbles — CHANGES_REQUESTED** [review]
I am not sure that we are properly setting psycopg2 up with unicode handling as described here? https://www.psycopg.org/docs/usage.html#unicode-handling

This might point to a way to handle this in a way that doesn't cause a huge performance regression.

Also, I think we should add a regression test for the specific case we are fixing here - a importing a unicode string for a node title that is at the max length.

File: kolibri/core/content/utils/channel_import.py (hunk 1 above)
Comment (jredrejo) [reply kolibri/core/content/utils/channel_import.py]: Changed it using a similar implementation of `execute_values`  but ensuring that utf-8 chars are inserted in the database and not bytes.

**Comment by jredrejo** [comment]
> Also, I think we should add a regression test for the specific case we are fixing here - a importing a unicode string for a node title that is at the max length.

Done

File: kolibri/core/content/fixtures/longdescriptions_content_data.json [hunk 2]
```diff
@@ -91,6 +91,10 @@
     {
       "id": "222455c2cc484298b1501a13e1c7eb5c",
       "tag_name": "tag_3"
+    },
+    {
+      "id": "0c20e2eb254b4070a713da63380ff0a3",
+      "tag_name": "velocidad de reacciones químicas"
```
Comment (rtibbles) [inline kolibri/core/content/fixtures/longdescriptions_content_data.json:101]: This looks like this is still ASCII rather than having any UTF-8 characters?

File: kolibri/core/content/utils/channel_import.py [hunk 3]
```diff
@@ -598,7 +599,14 @@ def postgres_table_import(self, model, row_mapper, table_mapper):
...
             for col_name, column_obj in columns:
                 default = self.get_and_set_column_default(column_obj)
-                value = row_mapper(record, column_obj.name)
+                value = row_mapper(record, col_name)
+                if not column_obj.primary_key and isinstance(
+                    column_obj.type, sa_String
+                ):
+                    max_length = column_obj.type.length
+                    if max_length is not None:
+                        value = value[:max_length] if value is not None else default
```
Comment (rtibbles) [inline kolibri/core/content/utils/channel_import.py:608]: Is this to handle cases where even with proper handling of utf-8 the tag name is too long because of slack of checks in generation of the SQLite db?

**Review by rtibbles — COMMENTED** [review]
Just a couple of questions to make sure the tests are doing what they ought - if the answer to my second question is yes, then we might need two test cases, one with utf-8 that is short enough but would overflow if it was converted to bytes, and another where the tag is just too long regardless.

I will manually test this to compare speed before and after.

File: kolibri/core/content/utils/channel_import.py (hunk 3 above)
Comment (jredrejo) [reply kolibri/core/content/utils/channel_import.py:608]: yes,  while testing this, I've seen tag names 40 chars long in sqlite while max_length is 30 in the model.

File: kolibri/core/content/fixtures/longdescriptions_content_data.json (hunk 2 above)
Comment (jredrejo) [reply kolibri/core/content/fixtures/longdescriptions_content_data.json:101]: yep, this test is to check 32 chars are cut to 30, just that. I used a real case from channel `07cd1633691b4473b6fda08caf826253` just for the _pleasure_ of using real data

Comment (jredrejo) [reply kolibri/core/content/fixtures/longdescriptions_content_data.json:101]: Added a new test _test_utf_test_keeps_its_length_ to check utf-8 strings are not byte encoded

**Comment by rtibbles** [comment]
This should be good to go, once I've done the speed test locally just to verify, will do tomorrow.

**Review by rtibbles — APPROVED** [review]
Local testing shows no performance regressions with these changes.
